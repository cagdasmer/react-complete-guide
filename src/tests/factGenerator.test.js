import React from "react";

import { shallow } from "enzyme";
import { wait } from "react-testing-library";
import axios from "axios";

import FactGenerator from "../components/factGenerator";

beforeEach(() => {
  jest.clearAllMocks();
  // setup
  axios.get.mockImplementationOnce(() =>
    Promise.resolve({
      data: {
        value: {
          joke: "Awesome fact!"
        }
      }
    })
  );
});

test("Should correctly render the FactGenerator component", () => {
  const component = shallow(<FactGenerator />);
  const fragment = component.instance().render();
  let wrapper = shallow(<div>{fragment}</div>);
  expect(wrapper.getElement()).toMatchSnapshot();
});

test("Message should change when the button is clicked", () => {
  const component = shallow(<FactGenerator />);
  component.find("button").simulate("click");
  expect(component.state("message")).toBe("Loading fact...");
  expect(component.state("loading")).toBe(true);
  const fragment = component.instance().render();
  let wrapper = shallow(<div>{fragment}</div>);
  expect(wrapper.getElement()).toMatchSnapshot();
});

test("Should correctly render the FactGenerator when the button is clicked", async () => {
  const component = shallow(<FactGenerator />);

  component.find("button").simulate("click");
  await wait(() => expect(component.state("loading")).not.toBe(true));
  expect(axios.get).toHaveBeenCalledTimes(1);
  expect(axios.get).toHaveBeenCalledWith("https://api.icndb.com/jokes/random");

  const fragment = component.instance().render();
  let wrapper = shallow(<div>{fragment}</div>);
  expect(wrapper.getElement()).toMatchSnapshot();
});
