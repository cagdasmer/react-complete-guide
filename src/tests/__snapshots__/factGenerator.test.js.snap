// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message should change when the button is clicked 1`] = `
<div>
  <React.Fragment>
    <div>
      Loading fact...
    </div>
    <button
      onClick={[Function]}
    >
      Load Fact
    </button>
  </React.Fragment>
</div>
`;

exports[`Should correctly render the FactGenerator component 1`] = `
<div>
  <React.Fragment>
    <div>
      No facts loaded yet
    </div>
    <button
      onClick={[Function]}
    >
      Load Fact
    </button>
  </React.Fragment>
</div>
`;

exports[`Should correctly render the FactGenerator when the button is clicked 1`] = `
<div>
  <React.Fragment>
    <_default
      text="Awesome fact!"
    />
    <button
      onClick={[Function]}
    >
      Load Fact
    </button>
  </React.Fragment>
</div>
`;
